<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GraphDialog</class>
 <widget class="QDialog" name="GraphDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Поиск кратчайшего пути в графе</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="vertexGroupBox">
     <property name="title">
      <string>Добавление вершины</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Номер вершины:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="vertexSpinBox">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>999</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addVertexButton">
        <property name="text">
         <string>Добавить вершину</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="edgeGroupBox">
     <property name="title">
      <string>Добавление ребра</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>От:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="fromVertexComboBox"/>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>До:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="toVertexComboBox"/>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Вес:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="weightSpinBox">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>999999.990000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addEdgeButton">
        <property name="text">
         <string>Добавить ребро</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="pathGroupBox">
     <property name="title">
      <string>Поиск пути</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>От:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="startVertexComboBox"/>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>До:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="endVertexComboBox"/>
      </item>
      <item>
       <widget class="QPushButton" name="findPathButton">
        <property name="text">
         <string>Найти путь</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QGroupBox" name="edgesGroupBox">
       <property name="title">
        <string>Список рёбер</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QListWidget" name="edgesListWidget"/>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="resultGroupBox">
       <property name="title">
        <string>Результат</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLabel" name="resultLabel">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="clearButton">
       <property name="text">
        <string>Очистить</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 